src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -564,76 +620,75 @@ async def ldap_validate(self, data, verrors):
-                'schema': 'RFC2307BIS',
-                'auxiliary_parameters': '\n'.join(aux_params)
-            })
-            return
+            default_naming_context = rootdse['defaultnamingcontext'][0]
+            data.update({'schema': 'RFC2307BIS'})
+            bases = data[constants.LDAP_SEARCH_BASES_SCHEMA_NAME]
+            bases[constants.SEARCH_BASE_USER] = f'cn=users,cn=accounts,{default_naming_context}'
+            bases[constants.SEARCH_BASE_GROUP] = f'cn=groups,cn=accounts,{default_naming_context}'
+            bases[constants.SEARCH_BASE_NETGROUP] = f'cn=ng,cn=compat,{default_naming_context}'
+            return constants.SERVER_TYPE_FREEIPA
 
-        elif 'domainControllerFunctionality' in rootdse[0]['data']:
+        elif 'domainControllerFunctionality' in rootdse:
             """
             ActiveDirectory domain.
             """
             self.logger.warning(
                 "Active Directoy plugin should be used for Active Directory domains."
             )
 
-            naming_ctx_dn = rootdse[0]['data']['rootDomainNamingContext'][0]
+            naming_ctx_dn = rootdse['rootDomainNamingContext'][0]
             naming_ctx = await self.get_dn(naming_ctx_dn, 'BASE', data)
             dom_sid_bytes = eval(naming_ctx[0]['data']['objectSid'][0])
             dom_sid = await self.object_sid_to_string(dom_sid_bytes)
+            maps = data[constants.LDAP_ATTRIBUTE_MAP_SCHEMA_NAME]
+            passwd = maps[constants.LDAP_PASSWD_MAP_SCHEMA_NAME]
+            group = maps[constants.LDAP_GROUP_MAP_SCHEMA_NAME]
             aux_params = [
-                'map passwd uid sAMAccountName',
-                'map passwd gidNumber primaryGroupID',
-                'map passwd gecos displayName',
                 'nss_uid_offset 20000',
                 'nss_gid_offset 20000',
-                f'map passwd uidNumber objectSid:{dom_sid}',
-                f'map group gidNumber objectSid:{dom_sid}',
-                f'filter passwd (sAMAccountType={SAMAccountType.SAM_USER_OBJECT.value})',
-                f'filter group (sAMAccountType={SAMAccountType.SAM_GROUP_OBJECT.value})',
             ]
+            passwd[constants.ATTR_USER_OBJ] = f'(sAMAccountType={SAMAccountType.SAM_USER_OBJECT.value})'
+            passwd[constants.ATTR_USER_NAME] = 'sAMAccountName'
+            passwd[constants.ATTR_USER_GID] = 'primaryGroupID'
+            passwd[constants.ATTR_USER_GECOS] = 'displayName'
+            passwd[constants.ATTR_USER_UID] = f'objectSid:{dom_sid}'
+            group[constants.ATTR_GROUP_OBJ] = f'(sAMAccountType={SAMAccountType.SAM_GROUP_OBJECT.value})'
+            group[constants.ATTR_GROUP_GID] = f'objectSid:{dom_sid}'
             data.update({
                 'schema': 'RFC2307BIS',
                 'auxiliary_parameters': '\n'.join(aux_params)
             })
-            return
+            return constants.SERVER_TYPE_ACTIVE_DIRECTORY
 
-        return
+        return constants.SERVER_TYPE_GENERIC
```

Comment:
Yeah, that's a good idea since this should only happen once.

Reply:
On second thought since I don't want to dump the full LDAP root DSE in a log message (maybe contains sensitive info), the log message would only reiterate info that's clearly visible in `ldap.config` output.